src/native/eventpipe/ep.h

```diff
@@ -105,6 +105,34 @@ ep_volatile_store_allow_write (uint64_t allow_write)
 	ep_rt_volatile_store_uint64_t (&_ep_allow_write, allow_write);
 }
 
+/*
+* EventPipeSessionOptions.
+*/
+
+#if defined(EP_INLINE_GETTER_SETTER) || defined(EP_IMPL_EP_GETTER_SETTER)
+struct _EventPipeSessionOptions {
+#else
+struct _EventPipeSessionOptions_Internal {
+#endif
+	const ep_char8_t *output_path;
```

Comment:
It was mainly to lessen the brain load between the options objects and the older `ep_enable` parameter's order that we can't change. I tried to follow [named struct initialization](https://gcc.gnu.org/onlinedocs/gcc-4.4.1/gcc/Designated-Inits.html#Designated-Inits): `struct point p = { .y = yvalue, .x = xvalue };` but the compiler was not happy about it. Maybe an init function that keeps the parameter order would be suitable?

Reply:
Understood, C99 designated initializers should work, we use it sparsely in Mono runtime since in the past we had an ambition to build all C code using C++ compiler, and C++ didn't pickup designated initializers until C++20, so we didn't use it. Doing a inline init method makes sense for the struct, we do similar for other structs that represent "objects", where each struct comes with a init/fini pair of functions. In this case, we can make an inline method together with the struct declare.

Reply:
I reordered the structure fields and created an init function but please check it's the way you intended. I'm not sure what is the best practice especially around fini method that would be empty in this case but it may be good to have it for completeness.

Reply:
Yes, the pattern has been that alloc/free comes in pair and init/fini comes in pair, just to make sure consumers call these functions in balanced pairs, so it will be possible to add things in the future that needs fini/free to be called without changing the usage pattern that should already been calling init/fini around usage of the type.